packages/mongodb-access-adapter/datagrip-access-adapter/src/main/kotlin/com/mongodb/jbplugin/accessadapter/datagrip/adapter/DataGripMongoDbDriver.kt

```diff
@@ -33,60 +37,90 @@ import kotlinx.coroutines.withTimeout
  */
 internal class DataGripMongoDbDriver(
     private val project: Project,
-    private val dataSource: LocalDataSource
+    private val dataSource: LocalDataSource,
 ) : MongoDbDriver {
     private val gson = Gson()
+    private val jsonWriterSettings = JsonWriterSettings.builder()
+.outputMode(JsonMode.EXTENDED)
+.indent(false)
+.build()
+
+    private fun Bson.toJson(): String = this.toBsonDocument().toJson(
+            jsonWriterSettings,
+        )
+
+    override suspend fun serverUri(): URI = URI.create(dataSource.url!!)
 
     override suspend fun <T : Any> runCommand(
-        command: Document,
+        command: Bson,
         result: KClass<T>,
-        timeout: Duration
-    ): T = withContext(
-        Dispatchers.IO
-    ) {
-        runQuery(
-            """db.runCommand(${command.toJson()})""",
-            result,
-            timeout
-        )[0]
-    }
+        timeout: Duration,
+    ): T =
+        withContext(
+            Dispatchers.IO,
+        ) {
+            runQuery(
+                """db.runCommand(${command.toJson()})""",
```

Comment:
We're using this for `buildInfo`, which I'm pretty sure always needs to run against `admin`. Should we explicitly pass the db here somewhere? It can still default to `admin`, but that should probably still be an argument to the Kotlin `runCommand()` wrapper

Reply:
True true, I'm adding the database to the runCommand wrapper.